streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -305,8 +305,21 @@ public MetadataResponse fetchMetadata() {
      * @throws StreamsException if brokers have version 0.10.0.x
      */
     public void checkBrokerCompatibility() throws StreamsException {
+        final String broker;
+        try {
+            broker = getAnyReadyBrokerId();
+        } catch (final StreamsException e) {
+            if ("Could not find any available broker.".equals(e.getMessage())) {
```

Comment:
We can introduce a new class `BrokerNotFoundException extends StreamsException`.